### PR TITLE
improved caching for geolocation service

### PR DESCRIPTION
**Pull request by hamidzr**
added precise caching for `nearbySearch` and `geolocate` RPCs

### CODE REVIEW

File: src/server/rpc/procedures/geolocation/geolocation.js [hunk 1]
```diff
@@ -33,7 +39,7 @@ if(!process.env.GOOGLE_GEOCODING_API) {
 
     // reverse geocoding helper, doesn't return a promise. handles sending of response.
     let reverseGeocode = (lat, lon, response, query)=>{
-        cache.wrap(lat + ', ' + lon + query, cacheCallback => {
+        cache.wrap(coordsToCacheKey(lat, lon) + query, cacheCallback => {
```
Comment (brollb) [inline src/server/rpc/procedures/geolocation/geolocation.js]: Why are you caching using the query? You should be able to cache the whole thing once across queries

File: src/server/rpc/procedures/geolocation/geolocation.js [hunk 2]
```diff
@@ -170,14 +176,16 @@ if(!process.env.GOOGLE_GEOCODING_API) {
             requestOptions.qs.keyword = keyword;
         }
 
-        return rp(requestOptions).then(res=>{
-            let places = res.results;
-            places = places.map(place => {
-                return [['latitude',place.geometry.location.lat],['longitude',place.geometry.location.lng],['name',place.name],['types',place.types]];
+        return cache.wrap(coordsToCacheKey() + keyword + radius, () => {
```
Comment (brollb) [inline src/server/rpc/procedures/geolocation/geolocation.js]: why aren't you passing the lat/long? Is this intentional?

Comment (hamidzr) [reply src/server/rpc/procedures/geolocation/geolocation.js]: nope, I'm adding some test..

File: src/server/rpc/procedures/geolocation/geolocation.js (hunk 1 above)
Comment (hamidzr) [reply src/server/rpc/procedures/geolocation/geolocation.js]: keeping the changes minimal. In addition, some parts are using are making the calls through a library and in other parts, the service manually makes the calls

**Pull request merged**